perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+- On Azure, we will use the MySQL service whenever it becomes availble.
+
+"""
+import json
+import logging
+import re
+import StringIO
+import time
+
+from perfkitbenchmarker import benchmark_spec as benchmark_spec_class
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.aws import aws_network
+
+FLAGS = flags.FLAGS
+flags.DEFINE_enum(
+    'db_instance_cores', '4', ['1', '4', '8', '16'],
+    'The number of cores to be provisioned for the DB instance.')
+
+flags.DEFINE_integer('oltp_tables_count', 4,
+                     'The number of tables used in sysbench oltp.lua tests')
+flags.DEFINE_integer('oltp_table_size', 100000,
+                     'The number of rows of each table used in the oltp tests')
+flags.DEFINE_integer('sysbench_warmup_seconds', 120,
+                     'The duration of the warmup run in which results are '
+                     'discarded.')
+flags.DEFINE_integer('sysbench_run_seconds', 480,
+                     'The duration of the actual run in which results are '
+                     'collected.')
+flags.DEFINE_integer('sysbench_thread_count', 16,
+                     'The number of test threads on the client side.')
+flags.DEFINE_integer('sysbench_latency_percentile', 99,
+                     'The latency percentile we ask sysbench to compute.')
+flags.DEFINE_integer('sysbench_report_interval', 2,
+                     'The interval we ask sysbench to report results.')
+
+BENCHMARK_INFO = {'name': 'mysql_service',
+                  'description': 'MySQL service benchmarks.',
+                  'scratch_disk': False,
+                  'num_machines': 1}
+
+# Query DB creation status once every 15 seconds
+DB_STATUS_QUERY_INTERVAL = 15
+
+# How many times we will wait for the service to create the DB
+# total wait time is therefore: "query interval * query limit"
+DB_STATUS_QUERY_LIMIT = 200
+
+AWS_PREFIX = ['aws', '--output=json']
+# Map from FLAGs.db_instance_cores to RDS DB Type
+RDS_CORE_TO_DB_CLASS_MAP = {
+    '1': 'db.m3.medium',
+    '4': 'db.m3.xlarge',
+    '8': 'db.m3.2xlarge',
+    '16': 'db.r3.4xlarge',  # m3 series doesn't have 16 core.
+}
+
+RDS_DB_ENGINE = 'MySQL'
+RDS_DB_ENGINE_VERSION = '5.6.23'
+RDS_DB_STORAGE_TYPE_GP2 = 'gp2'
+# Storage IOPS capacity of the DB instance.
+# Currently this is fixed because the cloud provider GCP does not support
+# changing this setting. As soon as it supports changing the storage size, we
+# will expose a flag here to allow caller to select a storage size.
+# Default GCP storage size is 1TB PD-SSD which supports 10K Read or 15K Write
+# IOPS (12.5K mixed).
+# To support 12.5K IOPS on EBS-GP, we need 4170 GB disk.
+RDS_DB_STORAGE_GP2_SIZE = '4170'
+
+# Constants defined for Sysbench tests.
+RAND_INIT_ON = 'on'
+DISABLE = 'disable'
+UNIFORM = 'uniform'
+OFF = 'off'
+MYSQL_ROOT_USER = 'root'
+MYSQL_ROOT_PASSWORD = 'Perfkit8'
```

Comment:
Lets still randomize this in case the current mode of operation changes or
its run on a VPC that has it exposed.  I'd prefer to be on the paranoid
side.

On Thu, Jul 23, 2015 at 1:46 PM, stfeng <notifications@github.com> wrote:

> In perfkitbenchmarker/benchmarks/mysql_service_benchmark.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/387#discussion_r35369232>
> :
>
> > +# Storage IOPS capacity of the DB instance.
> > +# Currently this is fixed because the cloud provider GCP does not support
> > +# changing this setting. As soon as it supports changing the storage size, we
> > +# will expose a flag here to allow caller to select a storage size.
> > +# Default GCP storage size is 1TB PD-SSD which supports 10K Read or 15K Write
> > +# IOPS (12.5K mixed).
> > +# To support 12.5K IOPS on EBS-GP, we need 4170 GB disk.
> > +RDS_DB_STORAGE_GP2_SIZE = '4170'
> > +
> > +# Constants defined for Sysbench tests.
> > +RAND_INIT_ON = 'on'
> > +DISABLE = 'disable'
> > +UNIFORM = 'uniform'
> > +OFF = 'off'
> > +MYSQL_ROOT_USER = 'root'
> > +MYSQL_ROOT_PASSWORD = 'Perfkit8'
>
> The DB is always created with public access disabled. Only the VMs inside
> the same VPC as the DB has access. That is controlled by IP level policy.
>
> For RDS, there is a password requirement (One capital letter + number)
> that's why I hand crafted this one. Otherwise, there is no need to specify
> a password for an ephemeral and test-only DB that is strictly only
> accessible by VMs inside the same VPC. ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/387/files#r35369232>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/
Benchmark like a pro... PerfKitBenchmarker
<https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>